File: src/main/java/com/matchFit/user/converter/StringToGenderConverter.java
package com.matchFit.user.converter;

import com.matchFit.user.entity.Gender;
import org.springframework.core.convert.converter.Converter;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Component;

@Component
public class StringToGenderConverter implements Converter<String, Gender> {

    @Override
    public Gender convert(@NonNull String source) {
        Gender gender = Gender.fromLabel(source);
        if (gender == null) {
            throw new IllegalArgumentException("Unknown gender: " + source);
        }
        return gender;
    }
}

File: src/main/kotlin/com/matchFit/user/converter/StringToGenderConverter.kt
package com.matchFit.user.converter

import com.matchFit.user.entity.Gender
import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component


@Component
class StringToGenderConverter : Converter<String, Gender> {
    override fun convert(source: String): Gender =
        Gender.fromLabel(source) ?: throw IllegalArgumentException("Unknown gender: $source")
}
